src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -273,7 +273,7 @@ private void BlockManagerOnSystemContractInvoked(object _, InvocationContext con
                 var tpkePublicKey =
                     PublicKey.FromBytes(args[1] as byte[] ?? throw new Exception("Failed to get tpkePublicKey"));
                 var tsKeys = new PublicKeySet(
-                    (args[2] as byte[][] ?? throw new Exception()).Select(x =>
+                    (args[3] as byte[][] ?? throw new Exception()).Select(x =>
```

Comment:
I think we should keep tskeys in args[2] and tpke keys at the last. New code tries to find tsKeys in args[3] and get error.